File: bms/go/go-edlib/StringHashMap_ToArray/StringHashMap_ToArray.go
package test

var ()

const ()

func (m StringHashMap) ToArray() []string {
	var index int
	arr := make([]string, 0, len(m))
	for key := range m {
		arr = append(arr, key)
		index++
	}

	return arr
}

// StringHashMap is HashMap substitute for string
type StringHashMap map[string]struct{}

File: translations_only/fail/go/go-edlib/StringHashMap_ToArray/mistral/CAPR/StringHashMap_ToArray.rs

use std::collections::HashMap;

fn to_array(m: HashMap<String, ()>) -> Vec<String> {
    let mut index = 0;
    let mut arr = Vec::new();
    for key in m.keys() {
        arr.push(key.to_string());
        index += 1;
    }

    return arr;
}

type StringHashMap = HashMap<String, ()>;
